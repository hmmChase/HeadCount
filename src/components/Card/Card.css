.aboveColor {
  background: rgb(152, 251, 152, 0.2);
}

.belowColor {
  background: rgb(219, 112, 147, 0.2)
}

.selectedCard {
  box-shadow: 0 8px 8px 0 rgba(102, 0, 0, 0.2);
}

.card {
  border-radius: 10px;
  border: 1px solid grey;
  cursor: pointer;
  width: 40vh;
  max-height: 56vh;
  margin: 2%;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  background-color: #ffffff63;
  cursor: pointer;
  transition: 0.3s;
  animation-duration: 2s;
  animation-name: card-opacity;
}

@keyframes card-opacity {
  from {
    opacity: 0.7;
  }
  to {
    opacity: 1;
  }
}

.card:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}

.card li {
  width: 28%;
  padding-left: 5px;
  margin: auto;
  text-align: justify;
}

.card ul {
  padding: 0;
  list-style-type: none;
}

.card h3 {
  padding: 2vh;
  margin-top: 7.2px;
  box-sizing: border-box;
  box-shadow: 1px 1px #1f1010;
  color: whitesmoke;
  background-color: #416387;
  text-align: center;
}

.clicked {
  border: 3px solid #f443368c;
  background-color: #ff57220d;
}

.clicked:hover {
  box-shadow: 1px 1px red;
}
