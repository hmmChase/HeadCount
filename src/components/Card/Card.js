import React from 'react';
import './Card.css';

const Card = props => {
  return <div>test</div>;
};

export default Card;
