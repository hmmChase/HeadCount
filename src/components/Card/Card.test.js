import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './Card';
// import { comparisonObject } from "../../__mocks__/mockData";

let mockProps = {
  location: 'lfjlsda',
  stats: {},
  changeSelectedDistricts: jest.fn(),
  clicked: false,
  average: 0.0343
};

describe('Card', () => {
  let card;
  beforeEach(() => {
    card = shallow(<Card {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(card).toMatchSnapshot();
  });

  it("not have a class of 'clicked' if the card has not been selected", () => {
    mockProps.clicked = false;

    expect(card.hasClass('clicked')).toEqual(false);
  });

  it("has a class of 'clicked' if the card has not been selected", () => {
    mockProps.clicked = true;
    card = shallow(<Card {...mockProps} />);

    expect(card.hasClass('clicked')).toEqual(true);
  });

  it('calls changeSelectedDistricts when card is clicked', () => {
    card = mount(<Card {...mockProps} />);
    card.find('.card').simulate('click');

    expect(card.prop('changeSelectedDistricts')).toHaveBeenCalledTimes(1);
  });
});
