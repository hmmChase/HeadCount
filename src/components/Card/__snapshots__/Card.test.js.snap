// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    average={0.034}
    changeSelectedDistricts={[Function]}
    clicked={false}
    location="lfjlsda"
    stats={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          lfjlsda
</h3>,
        <ul>
          
</ul>,
        false,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "lfjlsda",
        },
        "ref": null,
        "rendered": "lfjlsda",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
      false,
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            lfjlsda
</h3>,
          <ul>
            
</ul>,
          false,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "lfjlsda",
          },
          "ref": null,
          "rendered": "lfjlsda",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
        false,
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
